Builds/Test.py

```diff
@@ -0,0 +1,129 @@
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import argparse
+import os
+import platform
+import re
+import subprocess
+import sys
+
+IS_WINDOWS = platform.system().lower() == 'windows'
+
+if IS_WINDOWS:
+    BINARY_RE = re.compile(r'build\\([^\\]+)\\rippled.exe')
+
+else:
+    BINARY_RE = re.compile(r'build/([^/]+)/rippled')
+
+TEST_SUCCESS_MATCH = re.compile(r'\D0 failures')
+ALL_TARGETS = ['debug', 'release', 'all']
+
+parser = argparse.ArgumentParser(
+    description='Test.py - run ripple tests'
+)
+
+parser.add_argument(
+    '--verbose', '-v',
+    action='store_true',
+    help=('Report more information about which commands are executed and the '
+          'results.'),
+)
+
+parser.add_argument(
+    '--all', '-a',
+    action='store_true',
+    help=('Build all configurations.'),
+)
+
+parser.add_argument(
+    'scons_args',
+    default=(),
+    nargs='*'
+    )
+
+ARGS = parser.parse_args()
+
+def shell(*cmd, **kwds):
+    "Execute a shell command and return the output."
+    verbose = kwds.pop('verbose', ARGS.verbose)
+    if verbose:
+        print('$', ' '.join(cmd))
+    kwds['shell'] = IS_WINDOWS
+    processor = subprocess.call if verbose else subprocess.check_output
+    try:
+        return processor(cmd, **kwds)
+    except subprocess.CalledProcessError as e:
+        # Pretty up the exception a little with what you were doing.
+        raise ValueError('Couldn\'t execute "%s", errorcode=%s:\n%s' %
+                         (' '.join(cmd), e.returncode, e.output))
+
+if __name__ == '__main__':
+    args = list(ARGS.scons_args)
+    if ARGS.all:
+        for a in ALL_TARGETS:
+            if a not in args:
+                args.append(a)
+    print('Building:', *(args or ['(default)']))
+
+    # Build everything;  fail if this fails.
+    shell('scons', *args)
+
+    # Now extract the executable names and corresponding targets.
+    failed = []
+    lines = shell('scons', '-n', '--tree=derived', *args, verbose=False)
+    for line in lines.splitlines():
+        match = BINARY_RE.match(line)
+        if match:
+            executable, target = match.group(0, 1)
+            print('Testing', target)
+            test_result = shell(executable, '--unittest')
+            if TEST_SUCCESS_MATCH.find(test_result):
+                if ARGS.verbose:
+                    print(test_result)
+                try:
+                    npm = shell('npm', 'test', '--rippled=' + executable)
+                except Exception as e:
+                    if ARGS.verbose:
+                        print('ERROR:\n', e)
+                    failed.append([target, 'npm'])
+            else:
+                if ARGS.verbose:
+                    print('Unit test failure for', target)
+                    print(test_result)
+                failed.append([target, 'unittest'])
+    if failed:
+        print('FAILED:', *(':'.join(f) for f in failed))
+    else:
+        print('Success')
```

Comment:
I liked the behavior of the shell scripts which outputted which targets were built and tested. I figured it would prevent surprises if, for example, you passed a parameter expecting it to build "gcc.debug.nounity" as a dependency, and it didn't. Easy to miss with the output scrolling by.

Additionally, I don't see any kind of exit code, so I can't chain calls such as `Builds/Test.py all && build/msvc.debug/rippled ...`, which allows me to get rippled running automatically without babysitting if the tests all passed.